mainhandler/handlerequests.go

```diff
@@ -85,7 +87,15 @@ func (mainHandler *MainHandler) HandleWatchers(ctx context.Context) {
 		}
 	}()
 
-	watchHandler, err := watcher.NewWatchHandler(ctx)
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err)
+	}
+	ksStorageClient, err := kssc.NewForConfig(config)
```

Comment:
I will consider adding the storage client initialization to the `NewKubernetesApi()` function.

Reply:
Why?

It seems to me that the Storage client is pretty much stand-alone, not tied to the `KubernetesApi` interface, and very testable on its own. I am concerned that tying it to the KubernetesApi interface would make it harder to use Storage in tests while providing debatable additional benefits.